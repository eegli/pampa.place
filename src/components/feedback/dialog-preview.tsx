import {
  Box,
  Button,
  Dialog,
  DialogActions,
  DialogContent,
  DialogContentText,
  DialogTitle,
  useMediaQuery,
  useTheme,
} from '@mui/material';
import {ReactNode} from 'react';

type PreviewDialogProps = {
  onCloseCallback: () => void;
  children: ReactNode;
  title: string;
  text?: string;
};

export const PreviewDialog = (props: PreviewDialogProps) => {
  const theme = useTheme();
  const fullScreen = useMediaQuery(theme.breakpoints.down('sm'));
  const {text, title, children, onCloseCallback} = props;
  return (
    <Dialog
      open={true}
      fullScreen={fullScreen}
      onClose={onCloseCallback}
      PaperProps={{elevation: 1}}
      sx={{
        borderRadius: 10,
      }}
    >
      <DialogTitle>{title}</DialogTitle>
      <DialogContent
        sx={{
          display: 'flex',
          flexDirection: 'column',
        }}
      >
        <Box
          id={`preview-dialog`}
          sx={{
            height: 600,
            maxHeight: '100%',
            width: 400,
            margin: 'auto',
            maxWidth: '90%',
          }}
        >
          {children}
        </Box>
        {text && <DialogContentText p={2}>{text}</DialogContentText>}
      </DialogContent>
      <DialogActions>
        <Button onClick={onCloseCallback}>Close</Button>
      </DialogActions>
    </Dialog>
  );
};
