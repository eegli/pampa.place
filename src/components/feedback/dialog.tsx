import {
  Button,
  Dialog as MuiDialog,
  DialogActions,
  DialogContent,
  DialogContentText,
  DialogTitle,
} from '@mui/material';

export type DialogProps = {
  title: string;
  infoMessage: string;
  onCancelCallback: () => void;
  onConfirmCallback: () => void;
  confirmMessage?: string;
  cancelMessage?: string;
};

export const Dialog = ({
  title,
  infoMessage,
  onCancelCallback,
  onConfirmCallback,
  confirmMessage = 'Confirm',
  cancelMessage = 'Cancel',
}: DialogProps) => {
  return (
    <MuiDialog open PaperProps={{elevation: 1}}>
      <DialogTitle>{title}</DialogTitle>
      <DialogContent>
        <DialogContentText>{infoMessage}</DialogContentText>
      </DialogContent>
      <DialogActions sx={{mb: 2}}>
        <Button onClick={onCancelCallback} sx={{mr: 2}}>
          {cancelMessage}
        </Button>
        <Button
          variant="contained"
          onClick={onConfirmCallback}
          autoFocus
          sx={{mr: 2}}
        >
          {confirmMessage}
        </Button>
      </DialogActions>
    </MuiDialog>
  );
};
