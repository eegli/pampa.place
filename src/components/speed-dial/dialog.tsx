import {
  Button,
  Dialog,
  DialogActions,
  DialogContent,
  DialogContentText,
  DialogTitle,
} from '@mui/material';

export type DialogProps = {
  open: boolean;
  title: string;
  callback: () => void | (() => Promise<void>);
  callbackTitle: string;
  cancel: () => void;
  message?: string;
};

const ConfirmationDialog = ({
  open,
  title,
  callback,
  cancel,
  message,
  callbackTitle,
}: DialogProps) => {
  // Needed for smooth unmount transitions
  if (!open) {
    return <div />;
  }
  return (
    <div>
      <Dialog
        open={open}
        aria-labelledby="alert-dialog-title"
        aria-describedby="alert-dialog-description"
        PaperProps={{ elevation: 2 }}
      >
        <DialogTitle id="alert-dialog-title">{title}</DialogTitle>
        {message && (
          <DialogContent>
            <DialogContentText id="alert-dialog-description">
              {message}
            </DialogContentText>
          </DialogContent>
        )}
        <DialogActions>
          <Button onClick={cancel}>Cancel</Button>
          <Button onClick={callback} autoFocus>
            {callbackTitle}
          </Button>
        </DialogActions>
      </Dialog>
    </div>
  );
};

export default ConfirmationDialog;
