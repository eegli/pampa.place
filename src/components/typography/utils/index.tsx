import Box from '@mui/material/Box';

export const em = (txt: string, color: 'g' | 'v') => (
  <Box
    component="span"
    color={color === 'g' ? 'success.main' : 'secondary.main'}
  >
    <b>{txt}</b>
  </Box>
);
