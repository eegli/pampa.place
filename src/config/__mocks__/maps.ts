/* http://localhost:3000/api/maps/mocks */

import {generateMapData} from '../helpers/generator';
import {testMap} from '../__fixtures__/';

export const MAPS: ReturnType<typeof generateMapData> = new Map().set(
  testMap.properties.id,
  testMap
);
