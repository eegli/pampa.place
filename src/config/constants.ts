export const enum Constants {
  SESSION_API_KEY = 'gapikey',
  LOCALSTORAGE_MAPS_KEY = '@localmaps',
  THEME_KEY = 'theme',
}
