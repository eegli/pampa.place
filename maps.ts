import { CustomMaps } from '@/config/maps';

const maps: CustomMaps = [
  {
    type: 'FeatureCollection',
    features: [
      {
        type: 'Feature',
        properties: { name: 'Zurich' },
        geometry: {
          type: 'Polygon',
          coordinates: [
            [
              [8.530540466308594, 47.35696679330478],
              [8.576202392578125, 47.351384658074124],
              [8.576545715332031, 47.36231578548192],
              [8.568305969238281, 47.38789042338135],
              [8.545989990234375, 47.39904637835624],
              [8.490028381347656, 47.407179440027875],
              [8.472518920898438, 47.39253902496185],
              [8.473548889160156, 47.37277963653446],
              [8.507537841796875, 47.357431944587034],
              [8.530540466308594, 47.35696679330478],
            ],
          ],
        },
      },
    ],
  },
  {
    type: 'FeatureCollection',
    features: [
      {
        type: 'Feature',
        properties: { name: 'Frauenfeld' },
        geometry: {
          type: 'Polygon',
          coordinates: [
            [
              [8.893089294433594, 47.568708387119784],
              [8.871545791625977, 47.55886268619808],
              [8.875408172607422, 47.54403272736696],
              [8.901586532592773, 47.53557313731042],
              [8.915319442749023, 47.550926863012464],
              [8.91514778137207, 47.56812927946671],
              [8.893089294433594, 47.568708387119784],
            ],
          ],
        },
      },
    ],
  },
  /*  {
    type: 'FeatureCollection',
    features: [
      {
        type: 'Feature',
        properties: { name: 'Switzerland' },
        geometry: {
          type: 'Polygon',
          coordinates: [
            [
              [9.04449462890625, 45.82497145796607],
              [9.06097412109375, 45.91294412737392],
              [9.00604248046875, 45.960696964286164],
              [9.085693359375, 46.08275685027116],
              [9.23126220703125, 46.23305294479828],
              [9.28070068359375, 46.41892578708076],
              [9.31640625, 46.5002829039397],
              [9.42352294921875, 46.46813299215554],
              [9.5086669921875, 46.34503163945875],
              [9.64599609375, 46.30709840788667],
              [9.8272705078125, 46.356406479672486],
              [10.030517578125, 46.28052949433555],
              [10.09368896484375, 46.20644812194458],
              [10.1458740234375, 46.23305294479828],
              [10.11016845703125, 46.337447097476925],
              [10.17608642578125, 46.411351502899215],
              [9.99481201171875, 46.477590820859966],
              [10.0689697265625, 46.57774276255591],
              [10.22552490234375, 46.61548796222358],
              [10.349121093749998, 46.541860469382804],
              [10.458984375, 46.534303278597505],
              [10.4864501953125, 46.63435070293566],
              [10.3875732421875, 46.66451741754235],
              [10.4425048828125, 46.78501604269254],
              [10.491943359375, 46.9502622421856],
              [10.36285400390625, 46.97463048970666],
              [10.20355224609375, 46.89210855010362],
              [9.9371337890625, 46.92588289494367],
              [9.832763671875, 47.00460694243501],
              [9.57183837890625, 47.05702528260841],
              [9.46197509765625, 47.07012182383309],
              [9.459228515625, 47.18597932702905],
              [9.56085205078125, 47.32393057095941],
              [9.61029052734375, 47.455951443369926],
              [9.48394775390625, 47.585789182379905],
              [9.12139892578125, 47.66168780332917],
              [8.85498046875, 47.65613798222679],
              [8.80279541015625, 47.711609647845975],
              [8.6956787109375, 47.71530566159559],
              [8.71490478515625, 47.75225138174104],
              [8.646240234375, 47.787325537803106],
              [8.4759521484375, 47.748557989279426],
              [8.4210205078125, 47.67833372712059],
              [8.50616455078125, 47.63763431783611],
              [8.5858154296875, 47.646886969413],
              [8.5858154296875, 47.615421267605434],
              [8.492431640625, 47.60616304386874],
              [8.37432861328125, 47.56355410390806],
              [8.19580078125, 47.59875528481801],
              [7.9705810546875, 47.55243302404593],
              [7.74261474609375, 47.56726060598141],
              [7.583312988281249, 47.57652571374621],
              [7.48443603515625, 47.511635534978225],
              [7.371826171874999, 47.43366127871628],
              [7.18231201171875, 47.441092383072515],
              [7.176818847656249, 47.49864785970502],
              [6.99554443359375, 47.502358951968574],
              [6.9049072265625, 47.357431944587034],
              [7.020263671875, 47.349989032003215],
              [6.96258544921875, 47.28854494625741],
              [6.66595458984375, 47.060767479833494],
              [6.427001953125, 46.92588289494367],
              [6.43524169921875, 46.77749276376827],
              [6.11663818359375, 46.58906908309182],
              [6.053466796875, 46.443534945303284],
              [6.12762451171875, 46.36777895261494],
              [6.08642578125, 46.240651955001695],
              [5.9710693359375, 46.20264638061019],
              [5.96282958984375, 46.14178273759234],
              [6.089172363281249, 46.128459837044915],
              [6.229248046875, 46.1912395780416],
              [6.330871582031249, 46.24824991289166],
              [6.26495361328125, 46.30520105581194],
              [6.2841796875, 46.36967413462374],
              [6.47369384765625, 46.43217825300941],
              [6.8060302734375, 46.44164232762498],
              [6.7730712890625, 46.30899569419859],
              [6.80328369140625, 46.17792864897331],
              [6.94610595703125, 46.056079276178885],
              [7.0367431640625, 45.932050196856295],
              [7.40753173828125, 45.907211023476776],
              [7.6190185546875, 45.977877918928264],
              [7.797546386718749, 45.91294412737392],
              [7.901916503906249, 45.92822950933618],
              [8.06396484375, 46.05417324177816],
              [8.1463623046875, 46.18363372751015],
              [8.140869140625, 46.27863122156088],
              [8.308410644531248, 46.39051679686507],
              [8.41278076171875, 46.458673519829],
              [8.46771240234375, 46.26913887119721],
              [8.668212890625, 46.117037642576875],
              [8.8385009765625, 46.06560846138691],
              [8.887939453125, 45.9511496866914],
              [8.9373779296875, 45.838367585245855],
              [9.04449462890625, 45.82497145796607],
            ],
          ],
        },
      },
    ],
  }, */
];

export default maps;
