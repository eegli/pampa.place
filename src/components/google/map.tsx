import {svgMarker, svgMarkerColors} from '@/components/google/marker';
import {config} from '@/config/google';
import {LatLngLiteral, MapData} from '@/config/types';
import {Result} from '@/redux/game';
import {useAppDispatch} from '@/redux/hooks';
import {updateSelectedPosition} from '@/redux/position';
import {MapService, MarkerService, PolyLineService} from '@/services/google';
import {useEffect, useRef} from 'react';

export type GoogleMapProps = {
  map: MapData;
  mode?: 'preview' | 'play' | 'result';
  results?: Pick<Result, 'selected' | 'name'>[];
  initialPosition?: LatLngLiteral;
};

export const GoogleMap = ({
  map,
  mode,
  results,
  initialPosition,
}: GoogleMapProps) => {
  const dispatch = useAppDispatch();
  const ref = useRef<HTMLDivElement>(null);

  useEffect(() => {
    if (ref.current) {
      const unmount = MapService.mount(ref.current);
      /* Order in constructor is important! SW, NE  */
      const bounds = new google.maps.LatLngBounds(
        new google.maps.LatLng(map.properties.bbLiteral.SW),
        new google.maps.LatLng(map.properties.bbLiteral.NE)
      );
      google.maps.event.addListenerOnce(MapService.map, 'idle', () => {
        MapService.map.fitBounds(bounds, 0);
      });

      return () => {
        unmount();
      };
    }
  }, [map]);

  useEffect(() => {
    if (mode === 'preview') {
      console.info('PREVIEW MODE MOUNT');
      MapService.map.setOptions({
        ...config.map,
        gestureHandling: 'none',
        mapTypeId: 'roadmap',
        mapTypeControl: false,
      });

      const features = MapService.map.data.addGeoJson(map);

      MapService.map.data.setStyle({
        fillColor: '#003d80',
        fillOpacity: 0.2,
        strokeWeight: 0.8,
      });
      return () => {
        console.info('PREVIEW MODE UNMOUNT');
        features.forEach(feat => {
          MapService.map.data.remove(feat);
        });
      };
    }
  }, [mode, map]);

  useEffect(() => {
    if (mode === 'play') {
      console.info('PLAY MODE MOUNT');
      MapService.map.setOptions(config.map);

      const marker = new google.maps.Marker();
      marker.setMap(MapService.map);
      marker.setDraggable(true);

      MarkerService.add(marker);

      // This Google Map event is not typed unfortunately
      const listener = MapService.map.addListener('click', (e: unknown) => {
        const {latLng} = e as {latLng: google.maps.LatLng};
        marker.setPosition(latLng);
        dispatch(
          updateSelectedPosition({lat: latLng.lat(), lng: latLng.lng()})
        );
      });
      return () => {
        console.info('PLAY MODE UNMOUNT');
        if (listener) listener.remove();
        // https://developers.google.com/maps/documentation/javascript/markers#remove
        MarkerService.clearAllItems();
      };
    }
  }, [mode, dispatch]);

  useEffect(() => {
    if (mode === 'result' && initialPosition) {
      console.info('RESULT MODE MOUNT');
      MapService.map.setOptions(config.map);

      // Add original location marker
      const originMarker = new google.maps.Marker();
      originMarker.setPosition(initialPosition);
      originMarker.setMap(MapService.map);

      MarkerService.add(originMarker);

      // Add marker for each result
      results &&
        results.forEach((p, idx) => {
          if (p.selected) {
            const playerMarker = new google.maps.Marker({
              position: p.selected,
              map: MapService.map,
              label: {
                text: p.name,
                color: 'white',
                className: 'map-marker',
              },
              icon: {
                path: svgMarker.path,
                fillColor: `#${svgMarkerColors[idx]}`,
                fillOpacity: 1,
                anchor: new google.maps.Point(
                  svgMarker.anchor[0],
                  svgMarker.anchor[1]
                ),
                strokeWeight: 0,
                scale: 1,
                labelOrigin: new google.maps.Point(15, 60),
              },
            });

            const polyLine = new google.maps.Polyline({
              path: [initialPosition, p.selected],
              map: MapService.map,
              geodesic: true,
              strokeColor: `#${svgMarkerColors[idx]}`,
              strokeOpacity: 1.0,
              strokeWeight: 4,
            });
            PolyLineService.add(polyLine);
            MarkerService.add(playerMarker);
          }
        });

      return () => {
        console.info('RESULT MODE UNMOUNT');
        MarkerService.clearAllItems();
        PolyLineService.clearAllItems();
      };
    }
  }, [mode, initialPosition, results]);

  return (
    <div
      ref={ref}
      style={{
        height: '100%',
      }}
    />
  );
};
