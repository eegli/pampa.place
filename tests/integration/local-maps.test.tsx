import * as ConfirmDialog from '@/components/feedback/dialog';
import * as PreviewDialog from '@/components/feedback/dialog-preview';
import {Constants} from '@/config/constants';
import {LocalStorageMaps} from '@/config/types';
import * as helpers from '@/maps/helpers/parser';
import {userInputFeatureCollection} from '@/tests/fixtures/map';
import {act, fireEvent, render, screen} from '@/tests/utils';
import {MAPS} from 'src/maps';
import {MyMapsPage} from 'src/pages/my-maps.page';

beforeEach(() => {
  jest.clearAllMocks();
});

beforeAll(() => {
  jest.useFakeTimers();
  expect(MAPS.size).toBe(1);
});

afterAll(() => {
  jest.useRealTimers();
  expect(MAPS.size).toBe(1);
});

const inputs = {
  addMap: () => screen.getByRole('button', {name: /add map/i}),
  json: () => screen.getByLabelText(/geojson/i),
  name: () => screen.getByLabelText(/map name/i),
};

const spies = {
  dialog: jest.fn(), // jest.spyOn(ConfirmDialog, 'Dialog'),
  preview: jest.fn(), //jest.spyOn(PreviewDialog, 'PreviewDialog'),
  parser: jest.fn(), //jest.spyOn(helpers, 'parseUserGeoJSON'),
  setMap: jest.fn(), //jest.spyOn(MAPS, 'set').mockImplementation(jest.fn()),
  deleteMap: jest.fn(), // jest.spyOn(MAPS, 'delete').mockImplementation(jest.fn()),
};

function enterName(value: string) {
  fireEvent.change(inputs.name(), {
    target: {value},
  });
}

function enterJSON(value: unknown) {
  if (typeof value !== 'string') {
    value = JSON.stringify(value);
  }
  fireEvent.change(inputs.json(), {
    target: {value},
  });
  // The JSON input is debounced
  act(() => {
    jest.runOnlyPendingTimers();
  });
}

function getLocalMaps(): LocalStorageMaps {
  return JSON.parse(
    window.localStorage.getItem(Constants.LOCALSTORAGE_MAPS_KEY) || '{}'
  );
}

function expectDialogToBeGone() {
  expect(screen.queryByRole('dialog')).not.toBeInTheDocument();
}

describe('Integration, local map handling', () => {
  it.skip('map name input', () => {
    render(<MyMapsPage />);
    const input = inputs.json();
    expect(input).toHaveValue('');
    fireEvent.change(input, {
      target: {value: 'test'},
    });
    expect(input).toHaveValue('test');
  });
  it.skip('json input', () => {
    render(<MyMapsPage />);
    const input = inputs.json();
    enterJSON('asd');
    expect(input).toHaveValue('asd');
    // Make sure there is an error message
    expect(inputs.json()).toBeInvalid();
    enterJSON('{}');
    expect(input).toHaveValue('{}');
    expect(inputs.json()).not.toBeInvalid();
  });
  it.skip('parses user maps and adds them to global maps and local storage', () => {
    render(<MyMapsPage />);
    expect(
      screen.queryByRole('button', {name: /add map/gi})
    ).not.toBeInTheDocument();
    // Add invalid map
    enterName('this wont work');
    enterJSON('{}');
    fireEvent.click(inputs.addMap());
    expect(inputs.json()).toBeInvalid();
    // Add actual map
    enterName('my user map');
    enterJSON(userInputFeatureCollection);
    fireEvent.click(inputs.addMap());
    expect(inputs.json()).not.toBeInvalid();
    expect(spies.parser).toHaveBeenCalledWith(
      expect.any(String),
      'my user map',
      'local'
    );
    expect(spies.setMap).toHaveBeenCalledTimes(1);
    const localMaps = Object.values(getLocalMaps());
    expect(localMaps.length).toBe(1);
  });

  it.skip('allows previewing local maps', () => {
    const localMaps = Object.values(getLocalMaps());
    expect(localMaps.length).toBe(1);
    render(<MyMapsPage />);
    fireEvent.click(screen.getByRole('button', {name: 'preview-map-btn'}));
    expect(spies.preview).toHaveBeenCalledTimes(1);
    act(() => {
      spies.preview.mock.calls[0][0].onCloseCallback();
    });
    expectDialogToBeGone();
  });
  it.skip('allows editing local maps', () => {
    const localMaps = Object.values(getLocalMaps());
    expect(localMaps.length).toBe(1);
    render(<MyMapsPage />);
    expect(inputs.json()).toHaveValue('');
    fireEvent.click(screen.getByRole('button', {name: 'edit-map-icon'}));
    expect(inputs.json()).not.toHaveValue('');
  });
  it.skip('removes input maps from global maps and local storage', () => {
    const localMaps = Object.values(getLocalMaps());
    expect(localMaps.length).toBe(1);
    render(<MyMapsPage />);
    expect(screen.getAllByRole('listitem')).toHaveLength(1);
    fireEvent.click(screen.getByRole('button', {name: 'delete-map-icon'}));
    expect(spies.dialog).toHaveBeenCalledTimes(1);
    act(() => {
      spies.dialog.mock.calls[0][0].onCancelCallback();
    });
    // Canceling should clear the dialog
    expectDialogToBeGone();
    fireEvent.click(screen.getByRole('button', {name: 'delete-map-icon'}));
    // Delete the map from local storage and the global maps
    act(() => {
      spies.dialog.mock.calls[0][0].onConfirmCallback();
    });
    expectDialogToBeGone();
    expect(spies.deleteMap).toHaveBeenCalledTimes(1);
    const updatedLocalMaps = window.localStorage.getItem(
      Constants.LOCALSTORAGE_MAPS_KEY
    );
    expect(JSON.parse(updatedLocalMaps || '[]')).toStrictEqual({});
  });
});
