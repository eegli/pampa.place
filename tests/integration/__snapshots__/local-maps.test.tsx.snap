// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Integration, local map handling allows previewing local maps: map preview dialog 1`] = `
Object {
  "children": <GoogleMap
    map={
      Object {
        "geometry": Object {
          "coordinates": Array [
            Array [
              Array [
                8.530540466308594,
                47.35696679330478,
              ],
              Array [
                8.576202392578125,
                47.351384658074124,
              ],
              Array [
                8.576545715332031,
                47.36231578548192,
              ],
              Array [
                8.568305969238281,
                47.38789042338135,
              ],
              Array [
                8.545989990234375,
                47.39904637835624,
              ],
              Array [
                8.490028381347656,
                47.407179440027875,
              ],
              Array [
                8.472518920898438,
                47.39253902496185,
              ],
              Array [
                8.473548889160156,
                47.37277963653446,
              ],
              Array [
                8.507537841796875,
                47.357431944587034,
              ],
              Array [
                8.530540466308594,
                47.35696679330478,
              ],
            ],
          ],
          "type": "Polygon",
        },
        "properties": Object {
          "area": 34.542192851713864,
          "bb": Array [
            8.472518920898438,
            47.351384658074124,
            8.576545715332031,
            47.407179440027875,
          ],
          "bbLiteral": Object {
            "NE": Object {
              "lat": 47.407179440027875,
              "lng": 8.576545715332031,
            },
            "NW": Object {
              "lat": 47.407179440027875,
              "lng": 8.472518920898438,
            },
            "SE": Object {
              "lat": 47.351384658074124,
              "lng": 8.576545715332031,
            },
            "SW": Object {
              "lat": 47.351384658074124,
              "lng": 8.472518920898438,
            },
          },
          "category": "test",
          "id": "test-my-test-map",
          "name": "my test map",
        },
        "type": "Feature",
      }
    }
    mode="preview"
  />,
  "onCloseCallback": [Function],
  "text": "Rough bounds of the map \\"my test map\\"",
  "title": "Local map preview",
}
`;

exports[`Integration, local map handling removes input maps from global maps and local storage: map deletion dialog 1`] = `
Object {
  "confirmMessage": "Delete map",
  "infoMessage": "Are you sure you want to delete your local map \\"my test map\\"?",
  "onCancelCallback": [Function],
  "onConfirmCallback": [Function],
  "title": "Delete map",
}
`;
